### ISSUE TITLE
How to set a suitable batch size for video classification task of 32 frames?

### ISSUE THREAD

**Issue by realgump** [Sense-X/UniFormer#46]
I use 8 GPUs of 32G and set the batch size to 16, but there is always an out-of-memory error.

**Comment by Andy1621** [comment]
To save memory, you can use torch.utils.checkpoint via setting
```yaml
MODEL:
  USE_CHECKPOINT: True
  CHECKPOINT_NUM: [0, 0, 8, 0]
```
You can increase the number in `CHECKPOINT_NUM` to limit memory. 
Besides, when you use a smaller batch size, you should also decrease the learning rate.

**Comment by realgump** [comment]
I have used both checkpoint and mixed-precision but didn't work :(

**Comment by Andy1621** [comment]
Which model do you use? Have you tried to use `checkpoint` in all the stages? Like `[5, 8, 20, 7]`.

**Comment by realgump** [comment]
I tried [5, 8, 20, 7] and it did work, thanks a lot.
